gpMgmt/bin/gpload_test/gpload2/TEST_local_preload.py

```diff
@@ -68,3 +68,15 @@ def test_608_gpload_ext_staging_table():
     runfile(file)
     copy_data('external_file_13.csv','data_file.csv')
     write_config_file(reuse_tables=False, format='csv', file='data_file.csv', table='csvtable', delimiter="','", log_errors=True,error_limit=10,staging_table='"Staging_table"')
+
+@prepare_before_test(num=609, times=2)
+def test_609_gpload_fail_preload_truncate_rollback():
+    "609T gpload set preload truncate, but gpload fail, truncate should rollback together"
+    file = mkpath('setup.sql')
+    runfile(file)
+    f = open(mkpath('query609.sql'), 'w')
+    f.write("\\!  gpload -f "+mkpath('config/config_file')+"\n")
+    f.write("\\!  psql -d reuse_gptest -c \"SELECT count(*) from  testtruncate;\"\n")
+    f.close()
```

Comment:
It might be better to use Python's `with` statement to manage the file descriptor for us?

```suggestion
    with open(mkpath('query609.sql'), 'w') as f:
        f.write("\\!  gpload -f "+mkpath('config/config_file')+"\n")
        f.write("\\!  psql -d reuse_gptest -c \"SELECT count(*) from  testtruncate;\"\n")
```